RTC beta (#1778)
tinymce/tinymce-docs: rtc/getting-started.md

---
layout: default
title: Getting started with RTC
title_nav: Getting started
description: Getting started with RTC
keywords: rtc
---

> Requirements
> 1. Register for an API key at [{{site.accountpage}}]({{site.accountsignup}})
> 2. Generate a JWT key at [{{site.accountpage}}]({{site.accountpageurl}})
> 3. Store these keys temporarily, you will need them later.
> Do this before proceeding and it will go a lot smoother

## Simplest form of setup

> This is the most simple way to get the RTC plugin up and running using hard coded values. In a proper setup the document ID and secret would be retrieved from your server. This examples also assumes that a JWT provider endpoint exists at '/jwt' see the [JWT authentication]({{site.baseurl}}/rtc/jwt-authentication/) page for details on how to set that up.

Reviewer: Marking as resolved: Outdated before comment was added

```js
const yourDocumentId = 'some-document-id';
const yourSecretKey = 'your secret 🔑';

tinymce.init({
  selector: 'textarea',
  plugin: 'rtc',
  rtc_service_url: 'https://rtc.tiny.cloud/1/<your api key>',
  rtc_document_details_provider: () => Promise.resolve({ documentId: yourDocumentId }).
  rtc_encryption_provider: ({documentId, sessionId}) => {
    return Promise.resolve({ key: yourSecretKey });
  },
  rtc_token_provider: () => {
    return fetch('/jwt', {
      method: 'POST'
    });
  },
  rtc_snapshot: ({version, html}) => console.log(version, html)
})
```

#### Additional resources

> Go through the following additional resources to figure out how to configure RTC.

Reviewer: Marking as resolved: Outdated before comment was added

> - We recommend reading up and trying to understand how [JWT authentication]({{site.baseurl}}/rtc/jwt-authentication/) works.

Reviewer: Marking as resolved: Outdated before comment was added

- [Configuration Options]({{site.baseurl}}/rtc/configuration/)

## Need help? ##

> {{site.companyname}} is striving to make RTC as useful and simple as possible. For support related issues such as problems with JWT authentication and implementing RTC contact [{{site.supportname}}]({{site.supporturl}}).

Reviewer: Marking as resolved: Outdated before comment was added

---
layout: default
title: Getting started with RTC
title_nav: Getting started
description: Getting started with RTC
keywords: rtc
---

{% assign beta_feature = "The RTC plugin" %}
{% assign pre-release_type = "Closed Beta" %}
{% include misc/beta-note.md %}
{{site.requires_5_6v}}<br>
{{site.premiumplugin}}<br>

## Prerequisites

* A {{site.cloudname}} API key from [{{site.accountpage}}]({{site.accountsignup}}).

> * A JWT key generated at [{{site.accountpageurl}}]({{site.accountpageurl}}). Temporarily store the generated keys for use in the following procedure.

Reviewer 2: ```suggestion
* A JSON Web Token (JWT) key generated at [{{site.accountpageurl}}]({{site.accountpageurl}}). Temporarily store the generated keys for use in the following procedure.
```

This is the first time we use JWT on this page, so just expanding the acroymn.

## Basic RTC setup

> The following example shows how to get the RTC plugin up and running using hard coded values. For a production setup, the document ID and secret would be retrieved from your server. This examples also assumes that a JWT provider endpoint exists at '/jwt'. For information on setting up a JWT endpoint, see: [JWT authentication]({{site.baseurl}}/rtc/jwt-authentication/).

Reviewer 2: ```suggestion
The following example shows how to get the RTC plugin up and running using hardcoded values. For a production setup, the document ID and secret would be retrieved from your server. This example also assumes that a JWT provider endpoint exists at '/jwt'. For information on setting up a JWT endpoint, see: [JWT authentication]({{site.baseurl}}/rtc/jwt-authentication/).
```

```js
const yourDocumentId = 'some-document-id';

> const yourSecretKey = 'your secret 🔑';

Reviewer 2: Is it wise to use a unicode emoji here? This may not render well on some devices.
```suggestion
const yourSecretKey = 'your secret key';
```

Reviewer: no https://www.tiny.cloud/docs/configure/contributing-docs/#glossary

tinymce.init({
  selector: 'textarea',
  plugin: 'rtc',

>   rtc_document_details_provider: () => Promise.resolve({ documentId: yourDocumentId }).

Reviewer 2: @tylerkelly13 just wanted to double-check, but I'm guessing we're fine using ES6 examples here, given RTC isn't supported on IE 11?

Reviewer 2: Worth noting the highlighter doesn't seem to like using this syntax very much (or at least isn't highlighting the destructured object variables).

Reviewer: Correct. Although I prefer older syntax for readability purposes, this plugin won't work with IE11 and this helps to make that clear to people who skip the introduction section on supported browsers.

Re: syntax highlighting, yeah, seems it "gives up" when you have an object in a function in an object :man_shrugging:
